lib/src/shared/main/java/com/couchbase/lite/internal/core/C4Replicator.java

```diff
@@ -252,29 +252,33 @@ public C4ReplicatorStatus getStatus() {
         }
     }
 
-    public Set<String> getPendingDocIDs() throws LiteCoreException {
+    @Nullable
+    public byte[] getResponseHeaders() {
         synchronized (lock) {
-            if (handle == 0) { return null; }
-
-            final FLSliceResult result = new FLSliceResult(getPendingDocIds(handle));
-            try {
-                final FLValue slice = FLValue.fromData(result);
-                return (slice == null) ? Collections.emptySet() : new HashSet<>(slice.asTypedArray());
-            }
-            finally { result.free(); }
+            return (handle == 0) ? null : getResponseHeaders(handle);
         }
     }
 
-    public boolean isDocumentPending(String docId) throws LiteCoreException {
+    // Null return value indicates that this replicator is dead
+    @Nullable
+    public Boolean isDocumentPending(String docId) throws LiteCoreException {
         synchronized (lock) {
-            return (handle != 0) && isDocumentPending(handle, docId);
+            return (handle == 0) ? null : Boolean.valueOf(isDocumentPending(handle, docId));
         }
     }
 
+    // Null return value indicates that this replicator is dead
     @Nullable
-    public byte[] getResponseHeaders() {
+    public Set<String> getPendingDocIDs() throws LiteCoreException {
         synchronized (lock) {
-            return (handle == 0) ? null : getResponseHeaders(handle);
+            if (handle == 0) { return null; }
+
+            final FLSliceResult result = new FLSliceResult(getPendingDocIds(handle));
+            try {
+                final FLValue slice = FLValue.fromData(result);
+                return (slice == null) ? Collections.emptySet() : new HashSet<>(slice.asTypedArray());
+            }
+            finally { result.free(); }
```

Comment:
is it possible to get null result from JNI?

Reply:
Well... it is possible to get a `null` from `FLValue.fromData`.  That is not a JNI call...